plugins/modules/source_control/git/git_commit.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2020, Federico Olivieri (lvrfrc87@gmail.com)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+---
+module: git_commit
+author:
+    - "Federico Olivieri (@Federico87)"
+version_added: "2.10"
+short_description: Perform git and and git commit operations.
+description:
+    - Manage git add and git commit on local git repository.
+options:
+    path:
+        description:
+            - Folder path where .git/ is located.
+        required: true
+        type: path
+    comment:
+        description:
+            - Git commit comment. Same as "git commit -m".
+        type: str
+        required: true
+    add:
+        description:
+            - list of files to be staged. Same as "git add ."
+              Asterisx values not accepted. i.e. "./*" or "*".
+        type: list
+        default: ["."]
+        elements: str
+requirements:
+    - git>=2.10.0 (the command line tool)
+'''
+
+EXAMPLES = '''
+- name: Add and commit 2 files.
+  community.general.git_commit:
+    path: /Users/federicoolivieri/git/git_test_module
+    comment: My amazing backup
+    add: ['test.txt', 'txt.test']
+
+- name: Add all files using default and commit.
+  community.general.git_commit:
+    path: /Users/federicoolivieri/git/git_test_module
+    comment: My amazing backup
+'''
+
+RETURN = '''
+output:
+    description: list of git cli command stdout
+    type: list
+    returned: always
+    sample: [
+        "[master 99830f4] Remove [ test.txt, tax.txt ]\n 4 files changed, 26 insertions(+)..."
+    ]
+'''
+
+import os
+from ansible.module_utils.basic import AnsibleModule
+
+
+def git_commit(module):
+
+    commands = list()
+
+    add = module.params.get('add')
+    path = module.params.get('path')
+    comment = module.params.get('comment')
+
+    if add:
+        commands.append('git -C {path} add {add}'.format(
+            path=path,
+            add=' '.join(add),
+        ))
+
+    if comment:
+        commands.append('git -C {path} commit -m "{comment}"'.format(
+            path=path,
+            comment=comment,
+        ))
+
+    return commands
+
+
+def main():
+
+    argument_spec = dict(
+        path=dict(required=True, type="path"),
+        comment=dict(required=True),
+        add=dict(type="list", elements='str', default=["."]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+    )
+
+    result_output = list()
+    result = dict(changed=False)
+
+    for cmd in git_commit(module):
+        _rc, output, error = module.run_command(cmd, check_rc=False)
+
+        if output:
+            if 'no changes added to commit' in output:
+                module.fail_json(msg=output)
```

Comment:
Failing when there's nothing to commit isn't useful in all cases. How about making that configurable?

Reply:
Sure. I would you call the argument? `allow_empty` ?

Reply:
Hmm, wait, I think it is better to distinguish between three behaviors:
 - fail if nothing to commit;
 - don't commit if there's nothing to commit;
 - add empty commit if there's nothing to commit (git's `--allow-empty` option).

Maybe call the option `empty_commit` and make it an enum with values `fail`, `skip` and `empty_commit`. Or something similar.

(Now it also reminds me where I've seen this before: https://docs.buildbot.net/current/manual/configuration/buildsteps.html#gitcommit has an option `emptyCommits`. I remembered that I have actually added it because I needed it once :D Here's the PR: https://github.com/buildbot/buildbot/pull/5023 that might help you with implementing this.)

Reply:
It's ok. Good suggestion. That line you indicated though checks if there are not files to add. The commit check is the line below:

```
            elif 'nothing to commit, working tree clean' in output:
                module.fail_json(msg=output)
```

Reply:
Developing the `empty_commit=skip` I realize that is not really useful in my opinion. Basically, if I am not mistaken `skip` mode will allow the module just to stage file and personally I cannot see a use case where you want use a specific ansible module just to add files.

Reply:
`empty_commit=skip` means: if there's nothing to add (stage), don't commit.

Reply:
It all determines on what `git commit` does. Either it is potentially not run (`skip`), it gets the `--empty-commit` flag (`allow`), or it is run as usual (`fail`).